### PR TITLE
MAISTRA-1374: support multiple multus CNI interfaces

### PR DESCRIPTION
**Pull request by knrc**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -96,6 +96,7 @@ certmanager:
 #
 istio_cni:
   enabled: false
+  istio_cni_network: "istio-cni"
```
Comment (jwendell) [inline install/kubernetes/helm/istio/values.yaml:99]: If this is a new field (i.e., we don't need to keep api), why not just call it `network`? It's already within a `istio_cni` structure.

File: pkg/kube/inject/inject_test.go [hunk 2]
```diff
@@ -121,6 +122,19 @@ func TestIntoResourceFile(t *testing.T) {
 			readinessFailureThreshold:    DefaultReadinessFailureThreshold,
 			enableCni:                    true,
 		},
+		// verify cni network
```
Comment (jwendell) [inline pkg/kube/inject/inject_test.go:125]: I'm curious, in the case above, shouldn't the file `hello.yaml.cni.injected` contain the annotation with the default value?

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (knrc) [reply install/kubernetes/helm/istio/values.yaml:99]: It's not a new field, it's already in use within the operator

File: pkg/kube/inject/inject_test.go (hunk 2 above)
Comment (knrc) [reply pkg/kube/inject/inject_test.go:125]: No, the default values are not consumed in these tests

**Review by jwendell — APPROVED** [review]

File: install/kubernetes/helm/istio/files/injection-template.yaml [hunk 3]
```diff
@@ -453,6 +453,13 @@ dnsConfig:
     {{- end }}
 {{- end }}
 podRedirectAnnot:
+{{- if and (.Values.istio_cni.enabled) (ne .Values.istio_cni.istio_cni_network "") }}
+{{ if isset .ObjectMeta.Annotations `k8s.v1.cni.cncf.io/networks` }}
+   k8s.v1.cni.cncf.io/networks: "{{ index .ObjectMeta.Annotations `k8s.v1.cni.cncf.io/networks`}}, {{.Values.istio_cni.istio_cni_network}}"
```
Comment (dgn) [inline install/kubernetes/helm/istio/files/injection-template.yaml:458]: just out of curiosity: is the order important here? in my fix I had this the other way around

**Review by dgn — APPROVED** [review]

**Review by luksa — APPROVED** [review]

File: install/kubernetes/helm/istio/files/injection-template.yaml (hunk 3 above)
Comment (knrc) [reply install/kubernetes/helm/istio/files/injection-template.yaml:458]: I believe it is applied in order so the Istio CNI should be last to run

**Pull request merged**